supportedBackends/spark/src/main/scala/cromwell/backend/impl/spark/SparkClusterProcess.scala

```diff
@@ -0,0 +1,173 @@
+package cromwell.backend.impl.spark
+
+import java.nio.file.Path
+
+import akka.actor.ActorSystem
+import cromwell.backend.impl.spark.SparkClusterProcess.{ParserResponse, TerminalStatus}
+import spray.http.{HttpRequest, HttpResponse, StatusCodes}
+import spray.json.{DefaultJsonProtocol, JsonParser}
+import spray.client.pipelining._
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import better.files._
+import com.typesafe.scalalogging.Logger
+import org.slf4j.LoggerFactory
+import spray.httpx.unmarshalling._
+
+import scala.util.{Failure, Success, Try}
+
+object SparkClusterProcess {
+  sealed trait SparkClusterResponse
+  case class SuccessResponse(action: String, driverState: String, serverSparkVersion: String,
+                             submissionId: String, success: Boolean, workerHostPort: String, workerId: String) extends SparkClusterResponse
+  sealed trait ParserResponse
+  case class SuccessfulParsedResponse(action: String, message: String, serverSparkVersion: String, submissionId: String, success: Boolean) extends ParserResponse
+
+  sealed trait TerminalStatus
+  case class Failed(error: Throwable) extends TerminalStatus
+  case object Finished extends TerminalStatus
+
+  object SparkClusterJsonProtocol extends DefaultJsonProtocol {
+    implicit val sparkStatusResponseFormat = jsonFormat7(SuccessResponse)
+    implicit val successfulParsedResponseFormat = jsonFormat5(SuccessfulParsedResponse)
+  }
+}
+
+object SparkClusterConstants {
+  lazy val SparkClusterMasterDefault = "spark-master"
+  val HttpProtocol = "http://"
+  val SparkRestApiPort = "6066"
+  val SparkClusterIdentifier = Seq("6066", "spark")
+  val SparkClusterDeployMode = "cluster"
+  val PossibleSparkClusterMaster = Seq("spark", "Spark")
+  val SubmitJobJson = "%s.json"
+  val ReturnCodeFile = "rc"
+  val RunningStatus = "RUNNING"
+  val FinishedStatus = "FINISHED"
+  val FailedStatus = "FAILED"
+  val FAILED = -1
+  val FINISHED = 0
+  val sleepTime = 1000
+  val submissionIdInJson = "submissionId"
+  val successInJson = "success"
+  val submissionsApiLink = "v1/submissions/status"
+  lazy val SparkClusterMasterHostName = Try(sys.env("HOSTNAME")) match {
+    case Success(s) => Some(s)
+    case Failure(_) => None
+  }
+  val sparkClusterSubmissionResponseRegex = """(?s)\{(.*?)}""".r.unanchored
```

Comment:
I think the question mark in `.*?` is superfluous then; the `.*` will already match zero or more characters.

It's also possible that the capturing group should encompass the curly braces if you want to capture those too.